node/src/lib.rs

```diff
@@ -231,7 +232,12 @@ impl Node {
             });
         }
 
-        Ok(handles)
+        let (task_group, task_manager) = TaskGroup::new();
+        for (name, handle) in handles {
+            let _ = task_group.spawn(name, handle);
```

Comment:
what would be returned here?

Reply:
what are we spawning exactly?

Reply:
Added a comment. Basically a future that can be awaited as the JoinHandle is returned. But the returned future is unnecessary because we will use the `task_manager` to await for task terminations.